libs/s25main/buildings/nobBaseWarehouse.cpp

```diff
@@ -1026,15 +1026,14 @@ std::unique_ptr<nofDefender> nobBaseWarehouse::ProvideDefender(nofAttacker& atta
     // Kein Soldat gefunden, als letzten Hoffnung die Soldaten nehmen, die ggf in der Warteschlange noch hängen
     for(auto it = leave_house.begin(); it != leave_house.end(); ++it)
     {
-        nofSoldier* soldier;
+        std::unique_ptr<nofSoldier> soldier;
         // Soldat?
         if((*it)->GetGOT() == GO_Type::NofAggressivedefender)
         {
-            auto* aggDefender = static_cast<nofAggressiveDefender*>(it->get());
-            aggDefender->NeedForHomeDefence();
-            soldier = aggDefender;
+            soldier.reset(static_cast<nofSoldier*>(it->release()));
+            static_cast<nofAggressiveDefender&>(*soldier).NeedForHomeDefence();
```

Comment:
why not use `dynamic_pointer_cast` ?

Reply:
oh right, thats only for shared_ptrs. didnt we have something like `dynamic_unique_cast` already somewhere? I thought we have

Reply:
Found https://www.boost.org/doc/libs/1_64_0/libs/smart_ptr/pointer_cast.html and removed ours

Reply:
true I always forget that we have boost to our command